examples/MatchingRules_test.go

```diff
@@ -0,0 +1,92 @@
+package examples
```

Comment:
What is the difference between this test and the one in `GruleEngine_test.go` ? If its doing the same test, we wont need this one. Usually I put tests in `examples` folder for testing reported bug/issue, and to show the reporter that such bug/issue are no longer appear or not reproducable.